client/src/pages/private/SiteViewDetails.js

```diff
@@ -64,25 +97,29 @@ export default ({ match }) => {
         phases = await fetchSitePhases(site.id);
       }
       const participants = await fetchSiteParticipants(columnIDs, site.siteId);
+      const { hired, withdrawn } = await participants.json();
+      const hiredParticipants = mapSiteParticipantsDataToRow(hired, columnIDs);
+      const withdrawnParticipants = mapSiteParticipantsDataToRow(withdrawn, columnIDs);
       setIsLoading(false);
       return setSite({
         ...site,
-        hiredParticipants: participants.hiredParticipants,
-        withdrawnParticipants: participants.withdrawnParticipants,
+        hiredParticipants,
+        withdrawnParticipants,
         phases,
       });
     } else {
       openToast({
         status: ToastStatus.Error,
         message: response.error || response.statusText || 'Server error',
       });
-      return setSite({ hiredParticipants: [], withdrawnParticipants: [], phases: [] });
+      setIsLoading(false);
+      return setSite({});
     }
-  }, [id, setSite]);
+  }, [id, setSite, openToast, setIsLoading]);
 
   useEffect(() => {
     fetchDetails();
-  }, [id, fetchDetails]);
+  }, []);
```

Comment:
Another missing dependency here.

Reply:
I also added a check `if (site) fetchData()`, but due to setting the state multiple times in `fetchData`, the useEffect was being called multiple times, which I don't want. Removing the dependency resolved that - open to other solutions though!

Reply:
The fact that it does that feels a bit iffy to me, structurally - I'll see if I can find a fix.

Reply:
I have a potential solution, although it's more verbose.

1. Add a `hasFetched` state variable
2. In `fetchDetails`, add a `setHasFetched` call to show that we've tried to fetch at least once
3. Change the contents of lines 122-124 to:
```js
useEffect(() => {
    if (!hasFetched) fetchDetails();
  });
  ```

It's more verbose, but also makes it more obvious that we only want to try this once.

Reply:
Updated: Thanks for the suggestion, I did try it with the "isLoading" flag, but fetchData was updating it multiple times, and for some reason, I didn't think to create an additional flag just for this use case.

Reply:
Looks good now!